infoenergia_api/api/reports/reports.py

```diff
@@ -0,0 +1,40 @@
+from sanic import Blueprint
+from sanic.log import logger
+from sanic.response import json
+from sanic.views import HTTPMethodView
+from sanic_jwt.decorators import protected, inject_user
+
+from infoenergia_api.contrib import beedataApi, get_report_ids
+from infoenergia_api.contrib import PaginationLinksMixin
+
+
+bp_reports = Blueprint('reports')
+
+class ReportsView(PaginationLinksMixin, HTTPMethodView):
+    decorators = [
+        inject_user(),
+        protected(),
+    ]
+
+    endpoint_name = 'reports.reports'
+
+    async def post(self, request, user):
+        logger.info("Uploading contracts")
+        report_ids, month = await request.app.loop.create_task(get_report_ids(request))
+
+        unprocessed_reports = await request.app.loop.create_task(
```

Comment:
I think is not necessary awaiting process_reports function, it can be a background task. Think that until we don't process all contract in the request, beedata is waiting for a response of this endpoint.
I like think in this endpoint like this dialog:

* Bedata: hey som energia!, i have finished of calculate reports of this contract list
* SomEnergia: ok, i have received the contract list. Later, i will ask you for that reports.

Under that point of view, beedata only needs an ok response. Under the scenes, we will interrogate the beedata api to get the reports.

i propose you this change:
```python
request.app.loop.create_task(
   beedataApi().process_reports(report_ids, month)
 )
 response = {
     'reports': len(report_ids),
 }
 return json(response)
```